projects/fsastorefrontlib/src/cms-components/my-account/inbox/unread-messages-indicator/unread-messages-indicator.component.ts

```diff
@@ -0,0 +1,54 @@
+import { ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { AuthService, UserService } from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { InboxService } from '../../../../core/my-account/facade/inbox.service';
+
+@Component({
+  selector: 'cx-fs-unread-messages-indicator',
+  templateUrl: './unread-messages-indicator.component.html',
+})
+export class UnreadMessagesIndicatorComponent implements OnInit, OnDestroy {
+  numberOfUnreadMessages$: Observable<number>;
+  user$: Observable<boolean>;
+  subscription = new Subscription();
+
+  constructor(
+    protected inboxService: InboxService,
+    protected changeDetectorRef: ChangeDetectorRef,
+    protected authService: AuthService,
+    protected userService: UserService
+  ) {}
+
+  ngOnInit(): void {
+    this.user$ = this.authService.isUserLoggedIn();
+    this.subscription.add(
+      this.inboxService.unreadMessagesState
+        .pipe(
+          filter(isMessageRead => !!isMessageRead),
+          map(() => this.getNumberOfUnreadMessages())
+        )
+        .subscribe()
+    );
+    this.subscription.add(
+      this.userService
+        .get()
+        .pipe(
+          filter(user => Object.keys(user).length !== 0),
+          map(() => this.getNumberOfUnreadMessages())
+        )
+        .subscribe()
+    );
+  }
+
+  getNumberOfUnreadMessages() {
+    this.numberOfUnreadMessages$ = this.inboxService.getNumberOfUnreadMessages();
+    this.changeDetectorRef.detectChanges();
+  }
```

Comment:
Maybe there is no need for this method, if you are just setting this observable and if you unify the above mentioned subscriptions?

Reply:
Removed getNumberOfUnreadMessages method, but still need changeDetectorRef, because view doesn't refresh with new value even though I get new value in the component.

Reply:
Can we try to resolve all this in the HTML template?
```unreadMessagesState$ = this.inboxService.unreadMessagesState;
   this.user$ = this.userService.get().pipe(
          filter(user => Object.keys(user).length !== 0));
   this.numberOfUnreadMessages$ = this.inboxService.getNumberOfUnreadMessages()```
And then in the template nest them one inside other using async pipe for each observable?

Reply:
I'll try that. If that doesn't work I refactored code as you suggested to unify those two calls:
`    this.subscription.add(
      combineLatest([
        this.inboxService.unreadMessagesState$,
        this.userService.get()
      ])
        .pipe(
          filter(([isMessageRead, user]) => Object.keys(user).length !== 0),
          tap(() => {
            this.numberOfUnreadMessages$ = this.inboxService.getNumberOfUnreadMessages();
            this.changeDetectorRef.detectChanges();
          })
        )
        .subscribe()
    );`